src/t_hash.c

```diff
@@ -1134,6 +1134,20 @@ int hashTypeSetExInit(robj *key, robj *o, client *c, redisDb *db, const char *cm
         dictEntry *de = dbFind(c->db, key->ptr);
         serverAssert(de != NULL);
         lpt->key = dictGetKey(de);
+    } else if (o->encoding == OBJ_ENCODING_LISTPACK_EX) {
+        listpackEx *lpt = o->ptr;
+
+        /* If the hash previously had HFEs but later no longer does, the key ref
+         * (lpt->key) in the hash might become outdated after a MOVE/COPY/RENAME/RESOTRE
```

Comment:
maybe we should set lpt->key to NULL when it becomes trash?
i.e. if we no longer maintain it, and it risks being invalid, we can't trust it, and thus we better have it set to NULL and get NPD crash rather than something that occasionally works (hiding the bug).

Reply:
The ebuckets maintains trash flag on ebAdd, ebRemove and ebExpire.
We need to hook the corresponding places for in t_hash.c, for listpackex and 
dict that identify those changes. 

I prefer to do it on another commit due to time limit.